Build failure: ghostscript crosscompile (to armv7l) is broken

## Steps To Reproduce

Take the following flake.nix:

```
{
	inputs.nixpkgs-unstable.url = "github:nixos/nixpkgs/nixos-unstable";

	outputs = { self, nixpkgs-unstable, ... } @ inputs: let
		pkgs = import nixpkgs-unstable {
			localSystem = "x86_64-linux";
			crossSystem = "armv7l-linux";
			config = {
				allowUnsupportedSystem = true;
			};
		};
	in {
		legacyPackages = {
			"armv7l-linux".default = pkgs.stdenv.mkDerivation {
				pname = "test";
				version = "1.0";
				src = ./flake.nix;
				buildInputs = [ pkgs.ghostscript ];
			};
		};
	};
}
```

and run `nix build .#legacyPackages.armv7l-linux.default`

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/52ww7gl9i5a0sk885950n3ajr51dagdx-ghostscript-10.04.0.tar.xz
source root is ghostscript-10.04.0
setting SOURCE_DATE_EPOCH to timestamp 1726644855 of file ghostscript-10.04.0/zlib/zutil.h
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/knbr89jld24ya3malmlc7w8b1r3cjpw0-urw-font-files.patch
patching file Resource/Init/Fontmap.GS
Hunk #1 succeeded at 105 with fuzz 1 (offset 24 lines).
applying patch /nix/store/cfn0c1z68gw80d9d9pimy262xcqiz1lj-doc-no-ref.diff
patching file base/gs.mak
Hunk #1 succeeded at 561 (offset 23 lines).
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./freetype/builds/unix/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./lcms2mt/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./tiff/config/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./libpng/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./ijs/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./freetype/builds/unix/config.guess
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Updating Autotools / GNU config script to a newer upstream version: ./lcms2mt/config.guess
Updating Autotools / GNU config script to a newer upstream version: ./tiff/config/config.guess
Updating Autotools / GNU config script to a newer upstream version: ./libpng/config.guess
Updating Autotools / GNU config script to a newer upstream version: ./ijs/config.guess
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./freetype/builds/unix/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./lcms2mt/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./tiff/config/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./libpng/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./ijs/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./freetype/builds/unix/config.guess
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Updating Autotools / GNU config script to a newer upstream version: ./lcms2mt/config.guess
Updating Autotools / GNU config script to a newer upstream version: ./tiff/config/config.guess
Updating Autotools / GNU config script to a newer upstream version: ./libpng/config.guess
Updating Autotools / GNU config script to a newer upstream version: ./ijs/config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configure.ac:3413: warning: The macro 'AC_PROG_GCC_TRADITIONAL' is obsolete.
configure.ac:3413: You should run autoupdate.
./lib/autoconf/c.m4:1676: AC_PROG_GCC_TRADITIONAL is expanded from...
configure.ac:3413: the top level
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/p6k7xp1lsfmbdd731mlglrdj2d66mr82-bash-5.2p37/bin/sh"
configure flags: --prefix=/nix/store/qirj8929adyxn63z84k4cj1wqbv1igzz-ghostscript-with-X-armv7l-unknown-linux-gnueabihf-10.04.0 --bindir=/nix/store/qirj8929adyxn63z84k4cj1wqbv1igzz-ghostscript-with-X-armv7l-unknown-linux-gnueabihf-10.04.0/bin --sbindir=/nix/store/qirj8929adyxn63z84k4cj1wqbv1igzz-ghostscript-with-X-armv7l-unknown-linux-gnueabihf-10.04.0/sbin --includedir=/nix/store/qirj8929adyxn63z84k4cj1wqbv1igzz-ghostscript-with-X-armv7l-unknown-linux-gnueabihf-10.04.0/include --oldincludedir=/nix/store/qirj8929adyxn63z84k4cj1wqbv1igzz-ghostscript-with-X-armv7l-unknown-linux-gnueabihf-10.04.0/include --mandir=/nix/store/f4l5vvhw685qhajz990ly2bnq78njyb1-ghostscript-with-X-armv7l-unknown-linux-gnueabihf-10.04.0-man/share/man --infodir=/nix/store/qirj8929adyxn63z84k4cj1wqbv1igzz-ghostscript-with-X-armv7l-unknown-linux-gnueabihf-10.04.0/share/info --docdir=/nix/store/z84sbcm6xs1a287x4izmm7rdyqsxrbrx-ghostscript-with-X-armv7l-unknown-linux-gnueabihf-10.04.0-doc/share/doc/ghostscript-with-X-armv7l-unknown-linux-gnueabihf --libdir=/nix/store/qirj8929adyxn63z84k4cj1wqbv1igzz-ghostscript-with-X-armv7l-unknown-linux-gnueabihf-10.04.0/lib --libexecdir=/nix/store/qirj8929adyxn63z84k4cj1wqbv1igzz-ghostscript-with-X-armv7l-unknown-linux-gnueabihf-10.04.0/libexec --localedir=/nix/store/qirj8929adyxn63z84k4cj1wqbv1igzz-ghostscript-with-X-armv7l-unknown-linux-gnueabihf-10.04.0/share/locale --with-system-libtiff --without-tesseract --enable-dynamic --disable-hidden-visibility --with-x --enable-cups --build=x86_64-unknown-linux-gnu --host=armv7l-unknown-linux-gnueabihf
checking build system type... x86_64-unknown-linux-gnu
checking host system type... armv7l-unknown-linux-gnueabihf
checking for armv7l-unknown-linux-gnueabihf-gcc... armv7l-unknown-linux-gnueabihf-gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether armv7l-unknown-linux-gnueabihf-gcc accepts -g... yes
checking for armv7l-unknown-linux-gnueabihf-gcc option to enable C11 features... none needed
checking how to run the C preprocessor... armv7l-unknown-linux-gnueabihf-gcc -E
configure: Begin recursive call to configure script (for auxiliary tools)
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking for x86_64-unknown-linux-gnu-gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking how to run the C preprocessor... gcc -E
checking for a sed that does not truncate output... /nix/store/vmby25zqxnx94awvj0k7zn0rsd0zyrwg-gnused-4.9/bin/sed
checking for x86_64-unknown-linux-gnu-ranlib... armv7l-unknown-linux-gnueabihf-ranlib
checking for grep that handles long lines and -e... /nix/store/85amyk92rg19l4fy0qmy7wr4jmq8p5z0-gnugrep-3.11/bin/grep
checking for x86_64-unknown-linux-gnu-pkg-config... no
checking for pkg-config... no
checking for x86_64-unknown-linux-gnu-strip... no
checking for strip... /nix/store/4apajimszc47rxwcpvc3g3rj2icinl71-gcc-wrapper-13.3.0/bin/strip
checking if compiler supports restrict... yes
checking supported compiler flags...    -O2
   -DNDEBUG
   -Wall
   -Wstrict-prototypes
   -Wundef
   -Wmissing-declarations
   -Wmissing-prototypes
   -Wwrite-strings
   -fno-strict-aliasing
   -Werror=declaration-after-statement
   -fno-builtin
   -fno-common
   -Werror=return-type
   -Wno-unused-local-typedefs
   -gdwarf-2
   -g3
   -O0
   -fvisibility=hidden
 ...done.
checking compiler/linker address santizer support...  ...done.
checking compiler/linker address santizer build warnings support...  ...done.
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for vfork.h... no
checking whether byte ordering is bigendian... no
checking sse2 support... yes
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking for errno.h... yes
checking for fcntl.h... yes
checking for limits.h... yes
checking for malloc.h... yes
checking for memory.h... yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking for strings.h... (cached) yes
checking for sys/ioctl.h... yes
checking for sys/param.h... yes
checking for sys/time.h... yes
checking for sys/times.h... yes
checking for syslog.h... yes
checking for unistd.h... (cached) yes
checking for dirent.h... yes
checking for ndir.h... no
checking for sys/dir.h... yes
checking for sys/ndir.h... no
checking for inttypes.h... (cached) yes
checking for make... yes
checking if make is GNU make... yes
checking for sys/window.h... no
checking for an ANSI C-conforming const... yes
checking for inline... inline
checking for mode_t... yes
checking for off_t... yes
checking for size_t... yes
checking for struct stat.st_blocks... yes
checking whether struct tm is in sys/time.h or time.h... time.h
checking for dlopen in -ldl... yes
checking for dlfcn.h... yes
checking size of unsigned long long... 8
checking neon support... no
checking for cos in -lm... yes
checking for pread... yes
checking for pwrite... yes
checking for gcc options needed to detect all undeclared functions... none needed
checking whether pwrite is declared... yes
checking whether pread is declared... yes
checking whether popen is declared... yes
checking for pthread_create in -lpthread... yes
checking for recursive mutexes... PTHREAD_MUTEX_RECURSIVE
checking for local Tesseract library source... yes
checking whether the compiler supports GNU C++... yes
checking whether armv7l-unknown-linux-gnueabihf-g++ accepts -g... yes
checking for armv7l-unknown-linux-gnueabihf-g++ option to enable C++11 features... none needed
checking supported C++ compiler flags...    -std=c++17
-std=c++17
checking sse4.1 support... no
checking avx support... no
checking avx2 support... no
checking fma support... no
checking neon support... yes
checking for local zlib source... no
checking for deflate in -lz... no
configure: error: I did not find a copy of zlib on your system.
  Please either install it, or unpack a copy of the source in a
  local directory named 'zlib'. See http://www.gzip.org/zlib/
  for more information.

configure: error: Recursive call to configure script failed
```

</details>

## Additional context

ghostscript-with-X is a pretty common dependency of systems using X, so this makes cross compiling difficult for many systems.

The ghostscript docs mention that cross-compiling is only partially supported and requires special care: https://ghostscript.readthedocs.io/en/gs10.04.0/Make.html#cross-compiling

<!-- Add any other context about the problem here. -->

## Metadata

```
❯ nix-shell -p nix-info --run "nix-info -m"
fetching github input 'github:NixOS/nixpkgs/nixpkgs-unstable'
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/pvw05qnp5jiylq641k6scz71vr3rpwck-nix-info
copying path '/nix/store/pvw05qnp5jiylq641k6scz71vr3rpwck-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.1-arch1-1, Arch Linux, noversion, rolling`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/gilles/.config/nix/nix.conf:/home/gilles/.config/kdedefaults/nix/nix.conf:/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/lknx1arna0x37dg61ixsjbdlwcy9dvld-lix-2.91.1/share`
 - nixpkgs: `/nix/store/pzvsngvb3x5lnma3y855ij1m7s581dhm-source`
```

## Notify maintainers

@tobim 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/gh/ghostscript/package.nix
+++ b/pkgs/by-name/gh/ghostscript/package.nix
@@ -105,6 +105,14 @@ stdenv.mkDerivation rec {
     sed "s@if ( test -f \$(INCLUDE)[^ ]* )@if ( true )@; s@INCLUDE=/usr/include@INCLUDE=/no-such-path@" -i base/unix-aux.mak
     sed "s@^ZLIBDIR=.*@ZLIBDIR=${zlib.dev}/include@" -i configure.ac
 
+    # Sidestep a bug in autoconf-2.69 that sets the compiler for all checks to
+    # $CXX after the part for the vendored copy of tesseract.
+    # `--without-tesseract` is already passed to the outer ./configure, here we
+    # make sure it is also passed to its recursive invocation for buildPlatform
+    # checks when cross-compiling.
+    substituteInPlace configure.ac \
+      --replace-fail "--without-x" "--without-x --without-tesseract"
+
     autoconf
   '';
 

